server/api/inspect-url.post.ts

```diff
@@ -58,5 +64,51 @@ export default defineEventHandler(async (event) => {
 
   await saveRun(run)
 
-  return run
+  // Handle report creation/updating
+  let newReportId: string
+  const timestamp = Date.now()
+
+  if (currentReportId) {
+    try {
+      // Get existing report and create a new one with the additional run
+      const existingReport = await getReport(currentReportId)
+      const newRunIds = [...existingReport.runIds, run.runId]
+      newReportId = generateReportId(newRunIds, timestamp)
+
+      const newReport = {
+        reportId: newReportId,
+        runIds: newRunIds,
+        createdAt: timestamp,
+      }
+
+      await saveReport(newReport)
+    }
+    catch {
+      // If existing report not found, create a new one with just this run
+      newReportId = generateReportId([run.runId], timestamp)
+      const newReport = {
+        reportId: newReportId,
+        runIds: [run.runId],
+        createdAt: timestamp,
+      }
+
+      await saveReport(newReport)
+    }
```

Comment:
can you explain why we would do this? this sounds brittle and opaque. can't we just throw?